### PR TITLE
adds option to filter on private zone and fixes syntax.

### PR DESCRIPTION
**Pull request by tskolnik**
@brainsik This creates the option for creating a cert for a domain in a private hosted zone.

### CODE REVIEW

**Comment by gav-ashdown-onemedical** [comment]
Hey Jeremy - Theo works at One Med and we'd really like to get this merged.

@brainsik

File: main.tf [hunk 1]
```diff
@@ -46,7 +47,7 @@ resource "aws_route53_record" "main" {
 
 resource "aws_acm_certificate_validation" "main" {
   certificate_arn         = "${aws_acm_certificate.main.arn}"
-  validation_record_fqdns = aws_route53_record.main.*.fqdn
+  validation_record_fqdns = ["${aws_route53_record.main.*.fqdn}"]
```
Comment (dynamike) [inline main.tf]: This looks like 0.11 syntax. This module is now on 0.12 so the above change would break on 0.12.

File: main.tf [hunk 2]
```diff
@@ -23,6 +23,7 @@
 
 data "aws_route53_zone" "main" {
   name = "${var.zone_name}"
+  private_zone = "${var.private_zone}"
```
Comment (dynamike) [inline main.tf]: How is ACM able to validate the SSL certificate if the hosted zone is private? This module creates DNS records in the hosted zone that ACM can then validate to confirm the domain is in fact yours. Those records need to be accessible on the public internet to be validated.

Comment (tskolnik) [reply main.tf]: Yea, I actually ran into that very issue --though I was able to manually approve the certificate in the console (it was shown as pending approval), and terraform picks up on that approval and uses that for the validation resource in its state. We have a particular need at the moment to keep an application internal to the organization, thus our need for the private hosted zone. I copied the module locally and added the flag, still have some thinking to do about this. Thanks for your reply.

**Pull request closed without merging**